spring-data-jdbc/src/main/java/org/springframework/data/jdbc/core/convert/RowDocumentResultSetExtractor.java

```diff
@@ -66,9 +71,13 @@ static RowDocument toRowDocument(ResultSet resultSet) throws SQLException {
 		RowDocument document = new RowDocument(columnCount);
 
 		for (int i = 0; i < columnCount; i++) {
+
 			Object rsv = JdbcUtils.getResultSetValue(resultSet, i + 1);
 			String columnName = md.getColumnLabel(i + 1);
-			document.put(columnName, rsv instanceof Array a ? a.getArray() : rsv);
+			Object old = document.put(columnName, rsv instanceof Array a ? a.getArray() : rsv);
```

Comment:
I'm wondering why we're not using `putIfAbsent` semantics. Obtaining a value from the `ResultSet` typically uses the first occurrence of a column while we should be agnostic to the fact whether a column occurs multiple times in the result.

For reference, none of Spring JDBC's mappers care about duplicate columns, they just accept the fact that `ResultSet` either returns different values when iterating over the column count or use the value by name lookup.

Furthermore, when an external query declares the same column names, why is it us to validate this concern?

Reply:
FWIW, the JDBC spec says when you have duplicate column names, the first one should be returned.

Reply:
> why is it us to validate this concern?
Because we are nice people and we notice something is weird.